### ISSUE TITLE
Revisit use of mutex visitLock in Proxy

### ISSUE THREAD

**Issue by hferentschik** [fabric8-services/fabric8-jenkins-proxy#84]
The mutex protects _storageService.CreateStatistics_ which seems unnecessary. Does this not serialize all statistics creation?

**Comment by vpavlin** [comment]
I got some "database locked for write" errors (sadly, I did not record them properly) without the lock. My original assumption was that it should be DB to take care of this, but as I hit this, I just solved it with this..surely not optimal solution

**Comment by hferentschik** [comment]
> My original assumption was that it should be DB to take care of this

Right, that would be my assumption as well. Something is fishy here. Needs investigation.
